crates/containerd-shim-wasm/src/libcontainer_instance/mod.rs

```diff
@@ -1,4 +1,5 @@
 pub mod container_executor;
+pub mod easy;
```

Comment:
I am assuming that the `easy` mod is conditional compiled with `libcontainer` feature on, right?

Reply:
Yes, it is.

Reply:
If that's the case, shall we remove LibcontainerInstance trait in favor of this one?

Reply:
At the moment this API builds upon `LibcontainerInstance`.
As such both APIs can coexist.
If we decide we want the simplified API and there's no use case for `LibcontainerInstance` we can do that too.

Reply:
I think we need some kind of feedback from shim creators on that type of decision (even if its just us as we add support for other runtimes). It's a little extra work to maintain both but as it is here but having the ability to fall back on something that provides more flexibility seems like a good idea at least initially.  It seems like it would be pretty easy in the future to fold `LibContainerInstace` into `Engine `.

If we moved forward with this we would have three ways to implement shims, which all build on each other.

- `Instance` - you do everything
- `LibContainerInstance` - Provides ability to run container/wasm workloads side by side. You can configure Libcontainer.  note it is Linux Only.
- `Engine` - similar to LibContainerInstance but you don't have control over the  LibContainer.